src/amo/pages/Block/index.js

```diff
@@ -140,15 +114,40 @@ export class BlockBase extends React.Component<InternalProps> {
       return <ServerErrorPage />;
     }
 
-    const title =
-      block && block.name
-        ? i18n.sprintf(
-            i18n.gettext(`%(addonName)s has been blocked for your protection.`),
-            {
-              addonName: block.name,
-            },
-          )
-        : i18n.gettext(`This add-on has been blocked for your protection.`);
+    const isSoftBlocked =
+      block?.soft_blocked.length &&
+      (block?.soft_blocked.includes(match.params.versionId) ||
+        !block?.blocked.length);
```

Comment:
This should be a selector (e.g. `isSoftBlocked(guid, versionId)` defined in the reducer file) or at least computed in `mapStateToProps` IMO.